protostar/commands/test/test_cases.py

```diff
@@ -17,13 +19,47 @@ class TestCaseResult:
     def get_formatted_file_path(self):
         return log_color_provider.colorize("GRAY", str(self.file_path))
 
+    @abstractmethod
+    def display(self, include_stdout_section: bool = False) -> str:
+        ...
+
+
+@dataclass(frozen=True)
+class FinishedTestCase(TestCaseResult):
+    captured_setup_stdout: str
+    captured_test_stdout: str
```

Comment:
I think that it might be nice to have this model generalized over a possibility of having 0:* stdouts captured in test case. This way, it would be natural to write generic code over `setup` and `test` stages.

I'm thinking about following data model:

```python
class PassedTestCase:
    stdouts: List[OutputCapture]

@dataclass
class OutputCapture
    # typing pseudocode for brevity
    stream: "stdout" | "stderr" # possible future addition
    stage: str # "setup" or "test" but can be arbitrary
    data: bytes
```

It is also trivial to express interleaving of stdout/stderr ✖️ setup/test/whatever captures here. You just insert `OutputCapture`s for each captured line, one by one as encountered` to the list.

Reply:
I do wonder if we actually have to capture `setup` stdout ? I haven't ever needed such feature

Reply:
discussed privately, let's leave current implementation and consider this refactoring if we stumble upon a need for it